x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/BlockUtils.java

```diff
@@ -139,9 +139,14 @@ public static Block[] fromList(List<List<Object>> list) {
     }
 
     public static Block deepCopyOf(Block block) {
-        Block.Builder builder = block.elementType().newBlockBuilder(block.getPositionCount());
-        builder.copyFrom(block, 0, block.getPositionCount());
-        return builder.build();
+        return deepCopyOf(block, BlockFactory.getNonBreakingInstance());
+    }
+
+    public static Block deepCopyOf(Block block, BlockFactory blockFactory) {
```

Comment:
Probably worth a comment differentiating the two.

Reply:
added.